internal/plugin/connectors/tcp/mssql/connector.go

```diff
@@ -33,69 +118,102 @@ func (connector *SingleUseConnector) Connect(
 	}
 
+	// Prepare connection details from the client, formatted for MsSQL
 	connDetails, err := NewConnectionDetails(credentialValuesByID)
 	if err != nil {
-		connector.sendErrorToClient()
-		return nil, err
+		return connector.sendErrorWithMessage("Unable to create new connection details: %s", err)
 	}
 
-	err = connector.ConnectToBackend(dataSourceName(connDetails))
+	// Create a new MsSQL connector
+	// Using DSN (Data Source Name) string because gomssql forces us to.
+	//
+	// NOTE: Secretless has some unfortunate naming collisions with the
+	// go-mssqldb driver package.  The driver package has its own concept of a
+	// "connector", and its connectors also have a "Connect" method.
+	driverConnector, err := mssql.NewConnector(dataSourceName(connDetails))
 	if err != nil {
-		connector.logger.Errorf("Failed to connect to backend: %s", err)
-		connector.sendErrorToClient()
-		return nil, err
+		return connector.sendErrorWithMessage("Failed to create a go-mssqldb connector: %s", err)
 	}
 
-	// Now that the backend connection is authenticated, send the client
-	// a successful authentication response.
-	if _, err = clientConn.Write(connector.CreateAuthenticationOKMessage()); err != nil {
-		connector.logger.Errorf("Failed to send a successful authentication"+
-			" response to the client"+
-			": %s", err)
-		connector.sendErrorToClient()
-		return nil, err
-	}
+	// Create the context for our connection
+	parentContext := context.Background()
+	ch := make(chan map[uint8][]byte)
 
-	return connector.backendConn, nil
-}
+	// Set a 'marker' for when the driver has finished connecting
+	driverConnSet := make(chan struct {})
 
-// ConnectToBackend establishes the connection to the target database and sets
-// the backendConnection field.
-func (connector *SingleUseConnector) ConnectToBackend(dataSourceName string) error {
-	var err error
+	// Add channels to the context, to retrive information from the driver
+	childContext := context.WithValue(parentContext, PreLoginResponseKey, ch)
 
-	// Using DSN (Data Source Name) string because gomssql forces us to.
-	//
-	// NOTE: Secretless has some unfortunate naming collisions with the
-	// go-mssqldb driver package.  The driver package has its own concept of a
-	// "connector", and its connectors also have a "Connect" method.
-	driverConnector, err := mssql.NewConnector(dataSourceName)
+	// Build a new driver connection
+	var driverConn driver.Conn
+
+	go func() {
+		// Kick off authentication through our third party connector
+		driverConn, err = driverConnector.Connect(childContext)
+		driverConnSet <- struct{}{}
+
+		if err != nil {
+			_, err = connector.sendErrorWithMessage("failed to connect to mssql server: %s", err)
+		}
+	} ()
+
+	// Blocks continuation until we've received the preLoginResponse from the driver
+	preloginResponse := <- ch
+
+	// We don't want the driver to use encryption when talking to the server
+	preloginResponse[mssql.PreloginENCRYPTION] = []byte{mssql.EncryptNotSup}
```

Comment:
Isn't this one saying that we don't want the client to use encryption when talking to Secretless?

cc/ @doodlesbykumbi

Reply:
Correct. This is entirely a client to Secretless concern.